### PR TITLE
implement expression-limited fluxes (E-Flux like algorihm)

### LINKED ISSUE #552: Model type overlay that describes the genetic manipulation
Copied from #520:

Folding the logic into data is right. Actually I was contemplating on something similar already with #510.

Technically we can do an "overlay" type that just adds the functionality to the inner type, just as CoreModelCoupled adds the coupling or Serialized adds the disk cache.

Random idea:

```julia
struct WithGeneActivation{M :> MetabolicModel}
  m::M
  modified_genes::Dict{String,Float64}
end
```

_Originally posted by @exaexa in https://github.com/LCSB-BioCore/COBREXA.jl/issues/520#issuecomment-974107433_

### PR DESCRIPTION
**Pull request by exaexa**
Closes #552

I'll make a separate port to the 2.0 branch later.

### CODE REVIEW

**Comment by exaexa** [comment]
PS let this be the final feature of the 1.x branch. I'll release 1.5, the rest of the 1.5 milestone will be applied as bugfixes to 1.5.x or better to 2.x directly.

File: src/base/types/wrappers/ExpressionLimitedModel.jl [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+expression of genes (and thus the limited availability of the gene products).
+
+Use [`make_expression_limited_model`](@ref) or [`with_expression_limits`](@ref)
+to construct the models.
+
+E-Flux algorithm is closer described by: *Colijn, Caroline, Aaron Brandes,
+Jeremy Zucker, Desmond S. Lun, Brian Weiner, Maha R. Farhat, Tan-Yun Cheng, D.
+Branch Moody, Megan Murray, and James E. Galagan. "Interpreting expression data
+with metabolic flux models: predicting Mycobacterium tuberculosis mycolic acid
+production." PLoS computational biology 5, no. 8 (2009): e1000489*.
```
Comment (stelmo) [inline src/base/types/wrappers/ExpressionLimitedModel.jl:16]: People love making questionable cobra methods :/

File: src/base/utils/eflux.jl [hunk 2]
```diff
@@ -0,0 +1,16 @@
+
+"""
+$(TYPEDSIGNATURES)
+
+Create E-Flux-like bounds for a reaction that requires gene products given by
+`grr`, with expression "choke" data given by relative probabilities (between 0
+and 1) in `relative_expression`.
```
Comment (stelmo) [inline src/base/utils/eflux.jl:7]: interesting

File: src/base/utils/eflux.jl [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+"""
+function expression_probabilistic_bounds(
+    relative_expression::Dict{String,Float64},
+    grr::Maybe{Vector{Vector{String}}},
+)::Float64
+    isnothing(grr) && return 1.0
+    lup(g) = get(relative_expression, g, 1.0)
+    1 - prod(1 - prod(lup.(gr)) for gr in grr)
```
Comment (stelmo) [inline src/base/utils/eflux.jl:15]: Ah, I thought this would have some stochasticity  here

**Review by stelmo — APPROVED** [review]
Nice, 🚀

**Pull request merged**